roachtest, teamcity: publish runner logs to same directory

DarrylWong opened the pull request:
Previously we would publish the runner logs to the top level directory, causing it to be hard to find. This change publishes them to their own directory.

Fixes: #125298
Epic: none
Release note: none

DarrylWong committed:
roachtest: add always-collect-artifacts flag

This flag makes it so the test runner will collect artifacts during
test teardown, even if the test did not fail or time out.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/130963)
<!-- Reviewable:end -->

DarrylWong committed:
roachtest, teamcity: publish runner logs to same directory

Previously we would publish the runner logs to the top level
directory, causing it to be hard to find. This change publishes
them to their own directory.

DarrylWong commented:
Commit 1: Thoughts? I often want to grab the cockroach logs even if they didn't fail, but find it cumbersome to have to either use `debug-always` and ssh into the cluster or inject a failure. Not sure if I'm missing an easier way.

----
Commit 2:

<img width="736" alt="image" src="https://github.com/user-attachments/assets/77b292e1-26b8-4698-a3ae-2b88cf6cfd0e">

For more context: we publish the runner logs twice. Once invoked by the test runner and a second time by the teamcity job when it completes. The former is what's being changed in this PR, and what was causing the runner logs to be placed at the top level of the directory. For the record, GCE also had these top level files in addition to the ones found in `_runner-logs`.

Previously, the latter (teamcity job publishing artifacts) was what was actually putting the runner-logs in their own directory. This artifacts path was only configured for GCE, so I fixed that as well through the TC config.

herkolategan reviewed (approved):
Yes! Thanks, definitely something I've been wanting as well.

DarrylWong commented:
TFTR!

bors r=srosenberg,herkolategan
